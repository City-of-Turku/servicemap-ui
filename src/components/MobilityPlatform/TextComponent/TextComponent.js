import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import useLocaleText from '../../../utils/useLocaleText';

const TextComponent = ({
  classes, intl, textObj, isTitle, messageId,
}) => {
  const getLocaleText = useLocaleText();
  const wrapper = prop => (messageId ? intl.formatMessage({ id: messageId }, { value: prop }) : prop);
  return (
    <div className={classes.margin}>
      <Typography component="p" variant={isTitle ? 'subtitle1' : 'body2'}>
        {wrapper(getLocaleText(textObj))}
      </Typography>
    </div>
  );
};

TextComponent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  textObj: PropTypes.objectOf(PropTypes.any),
  isTitle: PropTypes.bool,
  messageId: PropTypes.string,
};

TextComponent.defaultProps = {
  textObj: {},
  isTitle: false,
  messageId: null,
};

export default TextComponent;
