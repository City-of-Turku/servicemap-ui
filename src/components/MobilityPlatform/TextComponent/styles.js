export default theme => ({
  margin: {
    margin: theme.spacing(0.4),
  },
  marginTop: {
    marginTop: theme.spacing(0.4),
  },
});
