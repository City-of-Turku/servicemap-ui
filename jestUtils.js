import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });