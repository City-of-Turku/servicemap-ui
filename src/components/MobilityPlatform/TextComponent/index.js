import { withStyles } from '@material-ui/core';
import TextComponent from './TextComponent';
import styles from './styles';

export default withStyles(styles)(TextComponent);
