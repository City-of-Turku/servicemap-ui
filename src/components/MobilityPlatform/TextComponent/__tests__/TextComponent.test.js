// Link.react.test.js
import React from 'react';
import TextComponent from '../index';
import { initialState } from '../../../../redux/reducers/user';
import { getRenderWithProviders } from '../../../../../jestUtils';

const mockProps = {
  textObj: {
    desc_fi: 'Testiteksti',
    desc_en: 'Test text',
    desc_sv: 'Test skrivning',
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<TextComponent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<TextComponent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<TextComponent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(mockProps.textObj.desc_fi);
  });
});
