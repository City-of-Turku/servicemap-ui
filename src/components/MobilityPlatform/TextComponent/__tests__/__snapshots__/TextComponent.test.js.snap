// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> should match snapshot 1`] = `
<div>
  <div
    class="css-1fhgjcy"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
    >
      Testiteksti
    </p>
  </div>
</div>
`;
